test/test_pipeline/components/preprocessing/test_imputers.py

```diff
@@ -66,31 +66,29 @@ def test_mean_imputation(self):
 
         # check if the fit dictionary X is modified as expected
         self.assertIsInstance(X['imputer'], dict)
-        self.assertIsInstance(categorical_imputer, BaseEstimator)
+        self.assertIsNone(categorical_imputer)
         self.assertIsInstance(numerical_imputer, BaseEstimator)
 
         # make column transformer with returned encoder to fit on data
-        column_transformer = make_column_transformer((categorical_imputer,
-                                                      X['dataset_properties']['categorical_columns']),
-                                                     (numerical_imputer,
+        column_transformer = make_column_transformer((numerical_imputer,
                                                       X['dataset_properties']['numerical_columns']),
                                                      remainder='passthrough')
         column_transformer = column_transformer.fit(X['X_train'])
         transformed = column_transformer.transform(data[test_indices])
 
-        assert_array_equal(transformed.astype(str), np.array([[1.0, 8.0, 9.0],
-                                                             [7.0, 3.5, 9.0],
-                                                             [4.0, 3.5, 3.0]], dtype=str))
+        assert_array_equal(transformed, np.array([[2.5, 8, 9],
+                                                  [7, 3.5, 9],
+                                                  [4, 3.5, 3]]))
 
     def test_median_imputation(self):
-        data = np.array([['1.0', np.nan, 3],
+        data = np.array([[1.0, np.nan, 3],
                          [np.nan, 8, 9],
-                         ['4.0', 5, np.nan],
+                         [4.0, 5, np.nan],
                          [np.nan, 2, 3],
-                         ['7.0', np.nan, 9],
-                         ['4.0', np.nan, np.nan]], dtype=object)
```

Comment:
Because we would like to test the median is correct, shouldn't we test by the data whose median is different from mean?